app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -80,6 +82,10 @@ class FxaRepo(
 
         @Suppress("DeferredResultUnused") // No value is returned & we don't need to wait for this to complete.
         accountManager.initAsync() // If user is already logged in, the appropriate observers will be triggered.
+
+        SendTabFeature(accountManager, pushRepo.pushFeature) { _, _ ->
```

Comment:
Can this get GC'd? Instead of doing the research, the quickest solution is to keep it in a local reference just to be safe (and comment why we do that).

Reply:
The default lifecycle for `SendTabFeature` is the whole application

Reply:
What do you mean by the default lifecycle?

I'm concerned that there may be no strong references to SendTabFeature and it may get garbage collected. a-c implements some of their listeners/observers with weak references so if those are the only references, SendTabFeature & it's observers may get GC'd.

Reply:
@mcomella The SendTabFeature accepts [a lifecycle owner](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/sendtab/src/main/java/mozilla/components/feature/sendtab/SendTabFeature.kt#L46) that defaults to [ProcessLifecycleOwner](https://developer.android.com/reference/android/arch/lifecycle/ProcessLifecycleOwner) which should result in the feature never getting GC'd unless the lifecycle owner is destroyed.

Reply:
Granted, it is a strange API we were trying to figure out how to reference during implementation since it doesn't need any references and only observes other features/managers.

Reply:
> The SendTabFeature accepts a lifecycle owner that defaults to ProcessLifecycleOwner which should result in the feature never getting GC'd unless the lifecycle owner is destroyed.

I wasn't sure that `ProcessLifecycleOwner` observers are guaranteed to not be GC'd but thinking back to how I used them in the past (as lambdas) and that the `Lifecycle.addObserver` method does not mention weak references, I now see that that should be fine. Additionally, our `ObserverRegistry.observers` keeps a strong reference to our observer.